### ISSUE TITLE
[Feedback] Ability to add docker labels

### ISSUE THREAD

**Issue by anlek** [IceWhaleTech/CasaOS#773]
### Description

I'd love to use Traefik to do proxy routings, however, the current version of CasaOS doesn't  support me adding/editing labels. Any chance that's a feature that's coming?

### Additional Information

	 - OS: linux
	 - CasaOS Version: 0.4.0
	 - Disk Total: 27712 
	 - Disk Used: 10471 
	 - System Info: [Debian GNU/Linux 11   \l Bit:64 Mem:7803M Core:4 Linux CasaOS 5.10.0-18-amd64 #1 SMP Debian 5.10.140-1 (2022-09-02) x86_64 GNU/Linux]
	 - Remote Version: 0.4.0
	 - Browser: Chrome 
	 - Version: 108.0.0.0

**Comment by tigerinus** [comment]
@anlek - We will be looking into this in future iteration.

Currently CasaOS depends on container labelling heavily, so exposing label editing is a bit risky. But we will see.

**Comment by gzavadskiy** [comment]
Ability to add docker labels, its very impotent please add it

**Comment by tigerinus** [comment]
It's such a shame that I haven't got to understand [how Traefik works](https://doc.traefik.io/traefik/getting-started/quick-start/) until recently... It's definitely a big plus to CasaOS.

Seriously looking into this...

**Comment by Jamyz** [comment]
I need label to use WATCHTOWER_LABEL_ENABLE="true"

**Comment by tigerinus** [comment]
> I need label to use WATCHTOWER_LABEL_ENABLE="true"

My bad this issue is meant to track labelling specific to Traefik.

While there is no UI to allow adding labels, you can export the compose file from your app as YAML, update the `label` property manually, and then go to terminal and run

```shell
casaos-cli app-management apply <appid> -f <path to the updated yaml file>
```

**Comment by mr-manuel** [comment]
As a workaround you can install and use Portainer in the meanwhile to edit the labels, if you don't want to do it over command line.

**Comment by anlek** [comment]
You could filter your own labels out of the "edit/add" list

**Comment by kforeverisback** [comment]
@tigerinus Looked at the CasaOS-App repo and CasaOS-UI repo. As far as I can tell, most of the App there doesn't expose some sort of common LABELS (they do expose some, but most of them are default labels from docker-image). And the CasaOS-UI repo, it doesn't seem its adding any extra casa specific Label either.

Could you point me to some codeblocks where I can take a look to see how the labels are used?

**Comment by hischampion** [comment]
Agreed this would be extremely helpful! Seems easy to filter special Casa labels by prefix? Alternatively only allow editing of labels that were originally user created, if a DB exists to remember that info.

Forgive my ignorance, please.

**Comment by anlek** [comment]
Any updates on the ETA on this ticket?

**Comment by Chuckame** [comment]
I am looking at moving out from cli oriented management, and this feature is for me a requirement to commit this migration.

I would propose another click-ful label edition:

Why not proposing some global configuration to set a default set of labels, env vars and more, where each param could be set as mandatory or not, with a meaningful name and description, default values based on the app name by example, and more.
And when you go to install/edit/update an app, it displays those pre-configured params in a form.

This would finally support any proxy (not only traefik) and all the docker-aware stuff for auth, monitoring and much more.

**Comment by StancuFlorin** [comment]
This missing functionality is a significant drawback for https://getwud.github.io/wud/#/
